src/plugins/vis_augmenter/public/saved_augment_vis/saved_augment_vis_references.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectAttributes, SavedObjectReference } from '../../../../core/public';
+import { AugmentVisSavedObject } from '../types';
+
+/**
+ * Note that references aren't stored in the object's client-side interface (AugmentVisSavedObject).
+ * Rather, just the ID/type is. The concept of references is a server-side definition used to define
+ * relationships between saved objects. They are visible in the saved objs management page or
+ * when making direct saved obj API calls.
+ *
+ * So, we need helper fns to construct & deconstruct references when creating and reading the
+ * indexed/stored saved objects, respectively.
+ */
+
+/**
+ * Used during creation. Converting from AugmentVisSavedObject to the actual indexed saved object
+ * with references.
+ */
+export function extractReferences({
+  attributes,
+  references = [],
+}: {
+  attributes: SavedObjectAttributes;
+  references: SavedObjectReference[];
+}) {
+  const updatedAttributes = { ...attributes };
+  const updatedReferences = [...references];
+
+  // Extract saved object
+  if (updatedAttributes.visId) {
+    updatedReferences.push({
+      name: 'visualization_0',
```

Comment:
Is `_0` meant to be auto-incrementing? Doesn't seem like something we want to hard-code, but I may be missing something.

Reply:
It's the standard for how saved objects inject/extract references. It could be auto-incrementing, in the case of [saved dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/dashboard/public/saved_dashboards/saved_dashboard_references.ts#L51) regarding panel references, or just hardcoded as 0 if there is a 1:1 mapping between a saved obj and a reference, like in [visualization saved objects](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/visualizations/public/saved_visualizations/saved_visualization_references.ts#L64) when saving a saved search.

Reply:
that all makes sense - can we add a comment to that effect?